pkg/utils/utils.go

```diff
@@ -76,7 +76,13 @@ func GetPfName(pciAddr string) (string, error) {
 		}
 		return "", err
 	} else if len(files) > 0 {
-		return files[0].Name(), nil
+		name, err := GetSriovnetOps().GetUplinkRepresentor(pciAddr)
```

Comment:
This will fail if device is not in switchdev mode.

we should imo.

1. check switchdev mode using devlink
2. if switchdev - use sriovnet
3. else use existing logic

Reply:
Agree to use the conditional check on switchdev mode.
It doesn't seem robust to return the right value based on a failure :-)

Reply:
Added a conditional check